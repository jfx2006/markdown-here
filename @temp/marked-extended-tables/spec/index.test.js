import { marked } from 'marked';
import extendedTable from '../src/index.js';

function trimLines(s) {
  return s.split('\n').map(l => l.trim()).join('\n');
}

describe('extended-table', () => {
  beforeEach(() => {
    marked.setOptions(marked.getDefaults());
  });

  test('Column Spanning', () => {
    marked.use(extendedTable());
    expect(marked(trimLines(`
      | H1      | H2      | H3      |
      |---------|---------|---------|
      | This cell spans 3 columns |||
    `))).toMatchSnapshot();
  });

  test('Row Spanning', () => {
    marked.use(extendedTable());
    expect(marked(trimLines(`
      | H1           | H2      |
      |--------------|---------|
      | This cell    | Cell A  |
      | spans three ^| Cell B  |
      | rows        ^| Cell C  |
    `))).toMatchSnapshot();
  });

  test('Multi-row headers', () => {
    marked.use(extendedTable());
    expect(marked(trimLines(`
      | This header spans two   || Header A |
      | columns *and* two rows ^|| Header B |
      |-------------|------------|----------|
      | Cell A      | Cell B     | Cell C   |
    `))).toMatchSnapshot();
  });
});
