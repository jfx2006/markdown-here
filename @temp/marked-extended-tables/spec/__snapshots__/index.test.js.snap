// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extended-table Column Spanning 1`] = `
"<table><thead><tr><th>H1</th>
<th>H2</th>
<th>H3</th>
</tr></thead><tbody><tr><td colspan=3>This cell spans 3 columns</td>
</tr></tbody></table>"
`;

exports[`extended-table Multi-row headers 1`] = `
"<table><thead><tr><th colspan=2 rowspan=2>This header spans two columns <em>and</em> two rows </th>
<th>Header A</th>
</tr><tr><th>Header B</th>
</tr></thead><tbody><tr><td>Cell A</td>
<td>Cell B</td>
<td>Cell C</td>
</tr></tbody></table>"
`;

exports[`extended-table Row Spanning 1`] = `
"<table><thead><tr><th>H1</th>
<th>H2</th>
</tr></thead><tbody><tr><td rowspan=3>This cell spans three  rows        </td>
<td>Cell A</td>
</tr><tr><td>Cell B</td>
</tr><tr><td>Cell C</td>
</tr></tbody></table>"
`;
